packages/odata-service-writer/src/index.ts

```diff
@@ -71,101 +155,146 @@ async function generate(basePath: string, service: OdataService, fs?: Editor): P
             }
         }
-
-        fs.write(paths.ui5Yaml, ui5Config.toString());
-
-        // ui5-local.yaml
-        ui5LocalConfigPath = join(dirname(paths.ui5Yaml), 'ui5-local.yaml');
-        if (fs.exists(ui5LocalConfigPath)) {
-            ui5LocalConfig = await UI5Config.newInstance(fs.read(ui5LocalConfigPath));
-            ui5LocalConfig.addFioriToolsProxydMiddleware({
-                backend: [service.previewSettings as ProxyBackend],
-                ignoreCertError: service.ignoreCertError
-            });
-        }
-    }
-
-    // Add mockserver entries
-    if (isServiceTypeEdmx && service.metadata) {
-        // mockserver entries are not required if service type is CDS
-        // copy existing `ui5.yaml` as starting point for ui5-mock.yaml
-        if (paths.ui5Yaml && ui5Config) {
-            const webappPath = await getWebappPath(basePath, fs);
-            const config = {
-                webappPath: webappPath,
-                ui5MockYamlConfig: { path: service.path }
-            };
-            await generateMockserverConfig(basePath, config, fs);
-            // add mockserver middleware to ui5-local.yaml
-            if (ui5LocalConfig) {
-                ui5LocalConfig.addMockServerMiddleware(service.path);
+        if (service.metadata) {
+            if (paths.ui5Yaml && ui5Config) {
+                const webappPath = await getWebappPath(basePath, fs);
+                const config = {
+                    webappPath: webappPath,
+                    ui5MockYamlConfig: { path: service.path, name: service.name }
+                };
+                // Generate mockserver middleware for ui5-mock.yaml
+                await generateMockserverConfig(basePath, config, fs);
+                // Update ui5-local.yaml with mockserver middleware from newly created/updated ui5-mock.yaml
+                await generateMockserverMiddlewareBasedOnUi5MockYaml(
+                    fs,
+                    paths.ui5Yaml,
+                    ui5LocalConfigPath,
+                    ui5LocalConfig
+                );
             }
-        }
-
-        // create local copy of metadata and annotations
-        fs.write(
-            join(basePath, 'webapp', 'localService', 'metadata.xml'),
-            prettifyXml(service.metadata, { indent: 4 })
-        );
-
-        // Adds local annotations to datasources section of manifest.json and writes the annotations file
-        if (service.localAnnotationsName) {
-            const namespaces = getAnnotationNamespaces(service);
-            fs.copyTpl(
-                join(templateRoot, 'add', 'annotation.xml'),
-                join(basePath, 'webapp', 'annotations', `${service.localAnnotationsName}.xml`),
-                { ...service, namespaces }
+            // Create local copy of metadata and annotations
+            fs.write(
+                join(basePath, 'webapp', 'localService', 'metadata.xml'),
```

Comment:
Question: above `webappPath` is read from project-access but here it's hard coded to `webapp`. Is this done on purpose?

Reply:
This one is from previous implementation. I guess I can move `const webappPath = await getWebappPath(basePath, fs);` above so it could be used in this case as well.